lib/rubocop/cop/style/redundant_array_flatten.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Checks for redundant calls of `Array#flatten`.
+      #
+      # `Array#join` joins nested arrays recursively, so flattening an array
+      # beforehand is redundant.
+      #
+      # @safety
+      #  Cop is unsafe because the receiver of `#flatten` might not
+      #  be an `Array`, so it's possible it won't respond to `#join`, or the end
+      #  result would be different.
```

Comment:
Can you indent it?

```suggestion
       #  Cop is unsafe because the receiver of `#flatten` might not
       #  be an `Array`, so it's possible it won't respond to `#join`, or the end
       #  result would be different.
```